src/app/utilities/transmart-utilities/transmart-constraint-mapper.ts

```diff
@@ -22,13 +22,24 @@ import {Study} from '../../models/constraint-models/study';
 import {Concept} from '../../models/constraint-models/concept';
 import {TrialVisit} from '../../models/constraint-models/trial-visit';
 import {TransmartConstraintSerialiser} from './transmart-constraint-serialiser';
-import {ConstraintHelper} from '../constraint-utilities/constraint-helper';
+import {SubselectionConstraint} from '../../models/constraint-models/subselection-constraint';
 
 export class TransmartConstraintMapper {
 
   static fullSerialiser = new TransmartConstraintSerialiser(true);
   static defaultSerialiser = new TransmartConstraintSerialiser(false);
 
+  public static mapConstraintOnPatientLevel(constraint: Constraint, full?: boolean): object {
+    let patientLevelConstraint: Constraint;
+    if (constraint.className === 'CombinationConstraint'
+      && (<CombinationConstraint>constraint).dimension === 'patient' || constraint.className === 'TrueConstraint') {
+      patientLevelConstraint = constraint;
+    } else {
+      patientLevelConstraint = new SubselectionConstraint('patient', constraint)
```

Comment:
Hm, this only seems to be relevant for the constraint mapper, so why not do this logic in the mapper, instead of introducing a new type here, that is only used for the mapper?

Reply:
it is also used for the combination constraint visitor, as you suggested here: https://github.com/thehyve/glowing-bear/pull/534#discussion_r262399883